### DISCUSSION TITLE
question about nbdcache

### DISCUSSION THREAD

**Discussion by marmara175** [chipsalliance/rocket-chip#3143] [Q&A]
Question1: 
ln line 742 of the file NBDcache.scala, can we change the code to the following code? 
> 	 s2_req.data := s1_req.data

Question2:
In line 739-740 of the NBDcache.scala, can we change the code to the following code? Because the s1_addr is physical.
> 	s2_req.phys := true
> s2_req.addr := s1_addr

Question3:
In NBDcache.scala, there are the registers s1_valid, s1_replay, s1_recycled, and the first two registers are connected to s2_valid and s2_replay, but there is no register named s2_recycled to connect with s1_recyled to control the recycled request can be dealt in stage3.

Question4:
ln line 904 of the file NBDcache.scala, why need a bubble?

**Comment by sequencer** [comment]
related to #2960
we need documentation on NBCache(I did a little, but that's quite primitive)
